src/client.ts

```diff
@@ -30,56 +24,39 @@ function eosPromisified(stream: NodeJS.ReadableStream | NodeJS.WritableStream):
 }
 
 export class MqttClient extends EventEmitter {
-  _incomingPacketParser: MqttParser;
-  _options: ConnectOptions;
-  disconnecting: any;
-  connecting: boolean;
-  connected: boolean;
-  errored: boolean;
-  _eos: Promise<void> | undefined;
-  conn: Duplex | Socket;
-  _clientLogger: Logger;
-  /**
-   * Use packet ID as key if there is one (e.g., SUBACK)
-   * Use packet type as key if there is no packet ID (e.g., CONNACK)
-   */
-  // TODO: This should be removed after we remove CONNECT into the sequencer
-  _inflightPackets: Map<string | number, (err: Error | null, packet: IPacket) => void>;
-  private _packetSequencer = new sequencer.MqttPacketSequencer(write.bind(null, this));
-
-  constructor(options: ConnectOptions) {
+  private _incomingPacketParser: MqttParser;
+  private _options: ClientOptions;
+  private _disconnecting: boolean;
+  private _connecting: boolean;
+  private _connected: boolean;
+  private _errored: boolean;
+  private _eos: Promise<void> | undefined;
+  private _conn: Duplex | Socket;
+  private _clientLogger: Logger;
+  private _packetSequencer: MqttPacketSequencer;
+
+  constructor(options: ClientOptions) {
     super();
     // assume the options have been validated before instantiating the client.
-    this.connecting = false;
-    this.connected = false;
-    this.errored = false;
-    this.disconnecting = false;
-    this._inflightPackets = new Map();
-
-    // Using this method to clean up the constructor to do options handling
-    logger.trace(`populating internal client options object...`);
-    this._options = {
```

Comment:
did you remove the default options?

Reply:
I moved it up to connect in index.ts because I feel like inserting default options and validating options should be done together